docs/src/main/asciidoc/spring-integration-pubsub.adoc

```diff
@@ -262,8 +262,10 @@ The provided `PubSubTemplate` contains all the necessary configuration to publis
 A publish timeout can be configured for synchronous publishing.
 If none is provided, the adapter waits indefinitely for a response.
 
-It is possible to set user-defined callbacks for the `publish()` call in `PubSubMessageHandler` through the `setPublishFutureCallback()` method.
-These are useful to process the message ID, in case of success, or the error if any was thrown.
+It is possible to set user-defined callbacks for the `publish()` call in `PubSubMessageHandler` through the `setSuccessCallback()` and `setFailureCallback()` methods (either one or both may be set).
+These give access to the Pub/Sub publish message ID in case of success, or the root cause exception in case of error.
```

Comment:
"message ID" always seems confusing to me. What we're really talking about is the "message ack ID", right?

Reply:
The confusion goes down to proto level. Publish operation returns [server-assigned message ID](https://github.com/googleapis/googleapis/blob/6afe7923cbf18ac9de4163fe1af316c74540b46b/google/pubsub/v1/pubsub.proto#L277). But when we acknowledge, it's called ["ack ID"](https://github.com/googleapis/googleapis/blob/6afe7923cbf18ac9de4163fe1af316c74540b46b/google/pubsub/v1/pubsub.proto#L1056).

Reply:
When you pull a message, you get an ack ID with it, which you can use to acknowledge it.
I don't think you get anything back when you acknowledge a message.

Reply:
Right, but we are not pulling a message. We are _publishing_ a message, so we are getting that message's ID back.